File: read_file_example.py
import sys
from pathlib import Path


def main() -> int:
    if len(sys.argv) != 2:
        print("Usage: python read_file_example.py <file-path>", file=sys.stderr)
        return 1

    file_path = Path(sys.argv[1])

    try:
        with file_path.open("r", encoding="utf-8") as file:
            for line in file:
                print(line, end="")
    except OSError as error:
        print(f"Could not read file: {error}", file=sys.stderr)
        return 1

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: ReadFileExample.java
import java.io.BufferedReader;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public class ReadFileExample {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.err.println("Usage: java ReadFileExample <file-path>");
            System.exit(1);
        }

        Path filePath = Paths.get(args[0]);

        try (BufferedReader reader = Files.newBufferedReader(filePath, StandardCharsets.UTF_8)) {
            String line;

            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }
        } catch (IOException error) {
            System.err.println("Could not read file: " + error.getMessage());
            System.exit(1);
        }
    }
}
